.github/workflows/scanner-build.yaml

```diff
@@ -14,14 +14,51 @@ on:
     - synchronize
 
 jobs:
+  define-job-matrix:
+    outputs:
+      matrix: ${{ steps.define-job-matrix.outputs.matrix }}
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+
+    - name: Define the matrix for build jobs
+      id: define-job-matrix
+      run: |
+        source './scripts/ci/lib.sh'
+
+        matrix='{ "build_and_push": { "name":[], "goos":[], "goarch":[] } }'
+
+        # The base matrix
+        matrix="$(jq '.build_and_push.name += ["default"]' <<< "$matrix")"
+        matrix="$(jq '.build_and_push.goos += ["linux"]' <<< "$matrix")"
+        # If this is updated, be sure to update architectures in push-manifests job below.
+        # TODO(ROX-21746): add arm64 back
+        matrix="$(jq '.build_and_push.goarch += ["amd64", "ppc64le", "s390x"]' <<< "$matrix")"
```

Comment:
Is there a specific reason not to initialize the matrix with the default name and 3 goos?

Reply:
I essentially copied this from `build.yaml` which did this, too. I can initialize it here, though